rust/src/aes.rs

```diff
@@ -0,0 +1,126 @@
+use crypto_pals::num_blocks;
+use hex::decode;
+use hex::encode;
+use openssl::error::ErrorStack;
+use openssl::symm::{decrypt, encrypt, Cipher};
+use rand::Rng;
+use std::io::Cursor;
+use std::str::from_utf8;
+
+const DEFAULT_KEY: &[u8] = b"YELLOW SUBMARINE";
+const BYTE_SIZE: usize = 8;
+
+pub struct MyCiphers {
+    core: Cipher,
+    key: Vec<u8>,
+    block_size: usize,
+}
+
+impl MyCiphers {
+    pub fn init_default() -> MyCiphers {
+        MyCiphers {
+            core: Cipher::aes_128_ecb(),
+            block_size: DEFAULT_KEY.len() * BYTE_SIZE,
+            key: DEFAULT_KEY.to_vec(),
+        }
+    }
+
+    pub fn init(key: Vec<u8>) -> MyCiphers {
+        MyCiphers {
+            core: Cipher::aes_128_ecb(),
+            block_size: key.len() * BYTE_SIZE,
+            key
+        }
+    }
+
+    pub fn getCore(&self) -> Cipher {
+        self.core
+    }
+
+    pub fn getKey(&self) -> &Vec<u8> {
+        &self.key
+    }
+
+    pub fn xcryptCtr(&self, text: &[u8], nonce: u64) -> Vec<u8> {
+        let mut counter: u64 = 0;
+        let keystream = self.generateKeystream(text, nonce);
+
+        keystream
+            .iter()
+            .zip(text.iter())
+            .map(|bytes: (&u8, &u8)| -> u8 { bytes.0 ^ bytes.1 })
```

Comment:
type inference can remove all the type annotations here
```suggestion
            .map(|(x, y)|x ^ y)
```

Reply:
Clutch